File: moveZeroes.cpp
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int j = 0;
        for (int i{}; i < nums.size(); i++)
        {
            if (nums[i] != 0)
                nums[j++] = nums[i];
        }
        for (; j < nums.size(); j++)
            nums[j] = 0;
    }
};

File: moveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        j=0
        
        for i in nums:
            if i!=0:
                nums[j]=i
                j+=1
                
        while(j<len(nums)):
            nums[j]=0
            j+=1

nums = [0, 0, 0, 2, 0, 1, 3, 4, 0, 0]
Solution().moveZeros(nums)
print(nums)
# [2, 1, 3, 4, 0, 0, 0, 0, 0, 0]